nomad/node_endpoint.go

```diff
@@ -1418,10 +1418,10 @@ func (n *Node) createNodeEvals(nodeID string, nodeIndex uint64) ([]string, uint6
 	// Create an evaluation for each system job.
 	for _, job := range sysJobs {
 		// Still dedup on JobID as the node may already have the system job.
-		if _, ok := jobIDs[job.ID]; ok {
+		if _, ok := jobIDs[*job.NamespacedID()]; ok {
```

Comment:
Out of scope: Should we drop the pointer from `Job.NamespacedID() *NamespacedID` so it matches the behavior of `structs.NewNamespacedID(...)` and `Allocation.JobNamespacedID()`?

Derefs without nil checks just always scare me even if they're "known" to be safe through manual code analysis.